### ISSUE TITLE
Add aerosol harness

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#461]
Build out the infrastructure for an online aerosol module (emissions, dry and wet deposition, activation with ARG). For now, mock out the actual microphysics which will be imported from external libraries

**Comment by duncanwp** [comment]
## Aerosol harness — plan (v2, MAM4-JAX-shaped)

Revised after looking at the concrete first microphysics core (**MAM4-JAX**) and the actual state of the "standardization" projects. Still the HAMMOZ-style split — a microphysics-agnostic **process harness** (emissions, dry/wet deposition, sedimentation, ARG activation) wrapping a swappable **microphysics core** — but the core *contract* is now shaped by MAM4-JAX reality and named in AMBRS's vocabulary, with part2pop demoted to an optional future diagnostics adapter.

### Why the change

| Project | What it actually is | Role here |
|---|---|---|
| [**MAM4-JAX**](https://github.com/reflective-org/MAM4-JAX) | JAX-native modal (4-mode) box microphysics. Ships its registry table in `mam4_jax/data.py` (modes, σ, density, κ, dgnum bounds, per-mode species, tracer-index maps). Owns intra-step micro; leaves emissions/deposition/sedimentation/**activation** OUT. | **The first real core.** Shape our contract around it. |
| [**AMBRS**](https://github.com/AMBRS-project/ambrs) | numpy ensemble *orchestrator*; runs MAM4 as a Fortran exe via namelists. Has its own modal structs (`AerosolModeState`, `AerosolSpecies` with κ/density/molar_mass). Only uses part2pop for post-run diagnostics. | **Naming source** — mirror its modal field vocabulary so a future adapter is cheap. |
| [**part2pop**](https://github.com/pnnl/part2pop) | numpy/scipy particle-resolved *analysis/diagnostics* lib. Non-differentiable (scipy root-finding, in-place mutation); no step/tendency contract; heavy deps. | **Future diagnostics adapter only.** Not a dependency. |

So part2pop can't be the swap point for a grad-through core — it's an analysis library, not a microphysics interface. The thing genuinely worth standardizing on is AMBRS's modal *vocabulary* plus MAM4-JAX's actual modal *layout*.

**Three concrete decisions vs the original plan:**
1. **No part2pop dependency.** Keep it as a later "export population → part2pop for CCN/optics" adapter. Nothing in the harness imports it.
2. **Registry is microphysics-neutral but MAM4-shaped, named in AMBRS vocab.** `HamRegistry`/`HamClass` → `ModalAerosolSpec` / `AerosolMode` / `AerosolSpecies` (`number`, `geom_mean_diam`, `geom_std_dev`, `mass_fractions`; species `molar_mass`/`density`/`hygroscopicity`).
3. **Placeholder core uses the real MAM4 4-mode layout** (`accum, aitken, coarse, primary_carbon`), built by wrapping MAM4-JAX's `data.py` constants. Swapping in real MAM4-JAX later replaces only the *step*, not the contract.

### MAM4-JAX realities we design around

- **Tracers:** interstitial **and a cloud-borne mirror** (MAM4 `q` + `qqcw`), plus per-mode number → tracer layout carries both (`m_/mc_/n_/nc_`).
- **Process split is clean:** core owns `calcsize`, water uptake, gas–aerosol exchange/condensation, nucleation, rename (Aitken→accum), coagulation, primary-carbon aging. Harness owns emissions, dry/wet dep, sedimentation, **activation** (MAM4-JAX's cloud-borne activation path isn't ported). Harness must not duplicate rename/aging.
- **Entry point:** `amicphys(state_dict, ...) -> state_dict` (one step = `calcsize → wateruptake → amicphys`), updated-state form.
- **Differentiability:** MAM4-JAX is jit/vmap-ready in spirit but grad-through is **not yet claimed** (bisection/regime branches; a `diffrax` migration is in progress upstream), and its driver is a Python loop, not `lax.scan`. → **The harness stays fully differentiable**; the core being grad-clean is a separate upstream milestone we don't block on. (Maintainer contact means we can request a `(state, env) -> state` / tendency-form entry point upstream if the box API needs a tweak.)

### Composability (unchanged shape)

Every harness step is a `PhysicsTerm` with declared `requires`/`provides`; the microphysics is one swappable term; aerosol mass+number (interstitial and cloud-borne) are normal `state.tracers` via `TracerSpec`, transported by dynamics. Scheme choice is a compose-time (Python) decision — no data-dependent shapes, no traced branching.

```python
def echam_physics(*, aerosol_module="macv2sp", ham_microphysics="placeholder", ...):
    ...  # "macv2sp" (default, unchanged) | "ham" -> ham_aerosol_physics(...) | "none"
```

ARG and SPA both write `activated_cdnc`, so the 2M micro reads whichever is composed in — toggling is one factory arg. MACv2-SP + SPA stay the default path, untouched.

### Phased delivery (PRs off `dev`)

| Phase | Scope | Acceptance |
|---|---|---|
| **0. Skeleton + population** | `population.py`, `species.py`, `mam4_data.py`, `tracer_layout.py`, `ham_state.py`, `microphysics/{base,placeholder}.py` | MAM4 spec round-trips; placeholder smoke-passes a column run; interstitial+cloud-borne tracers appear in `state.tracers` |
| **1. ARG activation** | `activation/{arg,arg_term}.py` | Within 1% of a `mo_ham_activ` port; `check_jvp` clean |
| **2. Sedimentation** | `sedimentation/sedi_term.py` | 1-D column mass conservation; gradients check |
| **3. Dry deposition** | `drydep/*`; expose `u_friction_velocity` | Per-mode v_dep within ~10% of Slinn/Ganzeveld; budget closes |
| **4. Emissions** | `emissions/*`; hook `ForcingData` sector fluxes + online seasalt/dust/DMS | Year-budget within ~30% of AeroCom; `check_vjp` clean |
| **5. Wet deposition** | `wetdep/*`; needs `precip_formation_rate` | In-cloud ∝ activated fraction; below-cloud size dependence; budget closes incl. re-evap |
| **6. Factory wiring** | `ham_terms.py`; `echam_physics(aerosol_module="ham")`; 2M reads `activated_cdnc` | Aquaplanet column run completes; tracers finite; 1-step gradient finite |
| **7. End-to-end** | `ham_integration_test.py` (T21 aquaplanet, placeholder + 2M) | CI pass; budget closure; no NaN; ARG vs SPA similar Nc; doc page |
| **8 (later, separate)** | Real MAM4-JAX core: `microphysics/mam4_jax.py` + tracer↔`q`/`qqcw` adapter | Forward column run finite; matches box fixture; grad test xfail pending upstream diffrax |

Out of scope: grad-clean MAM4 (upstream), SOA volatility basis, het-freezing, dust optics in radiation, the part2pop diagnostics adapter.

---

**Summary:** same HAMMOZ harness/core split, but the core contract is shaped by MAM4-JAX's actual 4-mode modal layout (incl. cloud-borne tracers) and named in AMBRS's vocabulary, so the population struct is the clean swap point. part2pop is an optional future diagnostics adapter, not a dependency. MACv2-SP + SPA remain the default; `aerosol_module="ham"` opts into the new harness with the κ-Köhler placeholder, and real MAM4-JAX drops in later by replacing only the microphysics term.

**Comment by duncanwp** [comment]
This [project](https://github.com/AMBRS-project/ambrs) looks a useful community effort which leverages standardize aerosol microphysics interfaces around https://github.com/pnnl/part2pop. Ideally we use this, or a compatible version, to define the populations and make microphysics more easily interchangable

**Comment by duncanwp** [comment]
Harness implemented
